src/rules/noUnboundMethodRule.ts

```diff
@@ -20,22 +20,55 @@ import * as ts from "typescript";
 import * as Lint from "../index";
 
 const OPTION_IGNORE_STATIC = "ignore-static";
+const OPTION_WHITELIST = "whitelist";
+
+const OPTION_WHITELIST_EXAMPLE = [
+    true,
+    {
+        [OPTION_WHITELIST]: ["expect", "typeof"],
+    },
+];
 
 interface Options {
     ignoreStatic: boolean;
+    whitelist: string[];
 }
 
 export class Rule extends Lint.Rules.TypedRule {
     /* tslint:disable:object-literal-sort-keys */
     public static metadata: Lint.IRuleMetadata = {
         ruleName: "no-unbound-method",
         description: "Warns when a method is used outside of a method call.",
-        optionsDescription: `You may optionally pass "${OPTION_IGNORE_STATIC}" to ignore static methods.`,
+        optionsDescription: Lint.Utils.dedent`
+            You may additionally pass "${OPTION_IGNORE_STATIC}" to ignore static methods, or an options object.
+            
+            The object might have two properties:
+            
+            * "${OPTION_IGNORE_STATIC}" - to ignore static methods.
+            * "${OPTION_WHITELIST}" - list of method names and keywords where method reference should be ignored, e.g.
+              * when method is referenced as a parameter to whitelisted method.
+              * when method is referenced in a typeof expression.
```

Comment:
> `"typeof"`

Love that this works in this PR! But - it seems a little odd to add `"typeof"` in the same array as method names. You can technically have `typeof` as a method name in some cases. Let's switch it to a separate member of the object. How about `allowTypeof`?

If you want to do that in a separate PR (so this one would start on the issue rather than completely fix it), that seems fine too.

Reply:
At first I wanted to make two whitelists for methods and operators, but I couldn't find any other operator that might be used in this context so I merged it to one parameter... But your suggestion feels best, so I added handling of `typeof` as a flag parameter.